crates/matrix-sdk-indexeddb/src/crypto_store.rs

```diff
@@ -1003,10 +1013,9 @@ impl_crypto_store! {
             )?
             .object_store(keys::SECRET_REQUESTS_BY_INFO)?
             .get(&self.encode_key(keys::KEY_REQUEST, key_info.as_key()))?
-            .await?
-            .and_then(|i| i.as_string());
+            .await?;
         if let Some(id) = id {
-            self.get_outgoing_secret_requests(id.as_str().into()).await
+            self.get_outgoing_secret_requests_from_encoded_key(id).await
```

Comment:
Don' t we need to encode the request ID here as well? If so, would it make sense to put the `self.encode_key()` call into the `get_outgoing_secret_request_from_encoded_key()` method?

Reply:
> Don' t we need to encode the request ID here as well?

No, I don't think so.

`id` here is pulled from the values stored in the `SECRET_REQUESTS_BY_INFO` store. That store is populated here: https://github.com/matrix-org/matrix-rust-sdk/blob/3d9cb19efbd4869c77fdadffc8de0052a5d20662/crates/matrix-sdk-indexeddb/src/crypto_store.rs#L608-L613

Note that the stored value (`key_request_id`) has already been passed through `encode_key`.